types/mysql/index.d.ts

```diff
@@ -101,6 +101,7 @@ interface IPoolCluster {
     add(group: string, config: IPoolConfig): void;
 
     end(): void;
+    end(callback: (err: IError, ...args: any[]) => void): void;
```

Comment:
This should be represented as an optional parameter, not two signatures with different arity.